tensorflow/compiler/xla/service/gpu/gpu_fusible.h

```diff
@@ -56,6 +56,9 @@ bool IsInputFusibleReduction(const HloInstruction& instr);
 bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
                                           const HloInstruction& instr2);
 
+// Whether `instr` is multi-output fusible
+bool IsMultiOutputFusible(const HloInstruction& instr);
```

Comment:
You need to use different functions for producer-consumer multi-output fusions.
For consumers you'd use this function, because it tells you whether an op can be the root of a multi-output fusion.
Producers need to work element-wise, i.e. you would use IsLoopFusible.
Currently this check is done here (and unnecessarily restricted to loop fusions):
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/multi_output_fusion.cc#L158